docs/env.md

```diff
@@ -1,117 +1,260 @@
 
-Here is a list of the ones that are needed or optional:
+First, there is our Django backend. We try to follow [12 factors app](https://12factor.net/) and so use environment variables for configuration.
 
-## DJANGO_SETTINGS_MODULE
+Then, there is our aws deployment configurations in the `src/aws` folder. It looks in `src/aws/env.d`.
 
-Description
+## 1. Django backend environment
 
-:   Define the settings file to use
+Specified in `{REPO_ROOT}/env.d/development`.
 
-Type
+### General Django settings
 
-:   String
+#### DJANGO_CONFIGURATION
 
-Mandatory
+Define the configuration to use in settings.
 
-:   Yes
+- Type: `String`
+- Required: Yes
+- Default: `Development`
+- Choices: Currently only `Development` is available.
 
-Default
+#### DJANGO_DEBUG
 
-:   None
+Turns on/off debug mode.
 
-Choices
+- Type: Boolean
+- Required: No
+- Default: `True` if `DJANGO_CONFIGURATION` is set to `Development`, `False` otherwise
+- Choices: `True` or `False`
 
-:   Must be set to `marsha.settings`
+#### DJANGO_SECRET_KEY
 
-## DJANGO_CONFIGURATION
+Standard Django secret key used to make the instance unique and generate hashes such as CSRF tokens or auth keys.
 
-Description
+- Type: String
+- Required: Yes
+- Default: None
 
-:   Define the configuration to use in settings
+#### DJANGO_SETTINGS_MODULE
 
-Type
+Define the settings file to use, relative to `src/backend`.
 
-:   String
+- Type: `String`
+- Required: Yes
+- Default: `marsha.settings`
+- Choices: Must be set to `marsha.settings`.
 
-Mandatory
+### Marsha specific settings
 
-:   Yes
+#### DJANGO_BYPASS_LTI_VERIFICATION
 
-Default
+Whether to skip all LTI-related checks and accept all requests on LTI endpoints as valid. This is useful for development but should never be set to `True` on publicly accessible deployments.
 
-:   None
+Note: Can't be set to `True` unless `DJANGO_DEBUG` is `True` too.
 
-Choices
+- Type: Boolean
+- Required: No
+- Default: `False`
+- Choices: `True` or `False`
 
-:   Currently only `Development` is available
+#### DJANGO_JWT_SIGNING_KEY
 
-## DJANGO_SECRET_KEY
+Secret key used to sign JWTs. Those are used to communicate between the Django backend and authenticated third parties (including the frontend).
 
-Description
+- Type: string
+- Required: Yes
+- Default: None
 
-:   Used to provide cryptographic signing, and should be set to a
-    unique, unpredictable value
+#### DJANGO_UPDATE_STATE_SHARED_SECRETS
 
-Type
+Secrets used to sign messages messages sent to the Django backend from AWS lambdas (like state updates). This being a list lets us support 1 or more shared secrets — and12 or more deployments — at the same time.
 
-:   String
+Note: should include the value from `TF_VAR_update_state_secret` in `src/aws` for any stack deployed to AWS which should communicate with the Django backend.
 
-Mandatory
+- Type: comma-separated list of strings
+- Required: Yes
+- Default: None
 
-:   Yes
+### Database-related settings
 
-Default
+#### DATABASE_URL
```

Comment:
actually this setting was removed and we are not using sqlite in development either.

As you can see [here](https://github.com/openfun/marsha/blob/master/src/backend/marsha/settings.py#L45), we now require separate settings for each connection parameter.

This reference to DATABASE_URL can be removed: https://github.com/openfun/marsha/blob/master/src/backend/marsha/settings.py#L27